File: src/FacilityManagement.Core/AggregatesModel/MaintenanceRequestAggregate/MaintenanceRequestDto.cs
namespace FacilityManagement.Core
{
    public class MaintenanceRequestDto
    {
        public Guid? MaintenanceRequestId { get; set; }
        public Guid? RequestedByProfileId { get; set; }
        public string? RequestedByName { get; set; }
        public DateTime Date { get; set; }
        public AddressDto? Address { get; set; }
        public string? Phone { get; set; }
        public string? Description { get; set; }
        public bool? UnattendedUnitEntryAllowed { get; set; }
        public Guid? ReceivedByProfileId { get; set; }
        public string? ReceivedByName { get; set; }
        public DateTime ReceivedDate { get; set; }
        public string? WorkDetails { get; set; }
        public DateTime WorkStarted { get; set; }
        public DateTime WorkCompleted { get; set; }
        public string? WorkCompletedByName { get; set; }
        public UnitEntered UnitEntered { get; set; }
        public ProfileDto? RequestedByProfile { get; set; }
        public MaintenanceRequestStatus Status { get; set; }
        public List<MaintenanceRequestCommentDto> Comments { get; set; } = new();
        public List<MaintenanceRequestDigitalAssetDto> DigitalAssets { get; set; } = new();
    }
}

File: src/FacilityManagement.App/src/app/@api/models/maintenance-request-dto.ts
/* tslint:disable */
import { AddressDto } from './address-dto';
import { MaintenanceRequestCommentDto } from './maintenance-request-comment-dto';
import { MaintenanceRequestDigitalAssetDto } from './maintenance-request-digital-asset-dto';
import { ProfileDto } from './profile-dto';
import { MaintenanceRequestStatus } from './maintenance-request-status';
import { UnitEntered } from './unit-entered';
export interface MaintenanceRequestDto {
  address?: AddressDto;
  comments?: Array<MaintenanceRequestCommentDto>;
  date?: string;
  description?: string;
  digitalAssets?: Array<MaintenanceRequestDigitalAssetDto>;
  maintenanceRequestId?: string;
  phone?: string;
  receivedByName?: string;
  receivedByProfileId?: string;
  receivedDate?: string;
  requestedByName?: string;
  requestedByProfile?: ProfileDto;
  requestedByProfileId?: string;
  status?: MaintenanceRequestStatus;
  unattendedUnitEntryAllowed?: boolean;
  unitEntered?: UnitEntered;
  workCompleted?: string;
  workCompletedByName?: string;
  workDetails?: string;
  workStarted?: string;
}
